### ISSUE TITLE
Variables inside language files

### ISSUE THREAD

**Issue by Arantor** [SimpleMachines/SMF#2664]
Can we _please_ remove all the variables from inside the language files except for $txt?

It's really not ideal and requires all kinds of convolutions in the language editor to support and can cause all kinds of mod related issues.

**Comment by Oldiesmann** [comment]
Which variables are used besides $context in a few spots?

**Comment by Arantor** [comment]
Looking at files that declare globals in them...

Admin uses $scripturl and declares $settings but never uses it
Drafts declares $scripturl and $context but uses neither
Editor uses its own language variable (that one's OK)
EmailTemplates declares $context, doesn't use it, declares $txtBirthdayEmails, not sure why
Errors uses $modSettings and $scripturl
Help uses $helptxt (that one's OK) and uses function_exists() calls inline to determine between different strings
Login uses $context
ManageMembers declares $context but doesn't use it
ManagePaid uses $boardurl
ManageSettings uses $scripturl
Modlog declares $scripturl but never uses it
PersonalMessage declares $context but never uses it
Post declares $context but never uses it
Profile uses $scripturl and $context
Settings uses $settings (I'd argue that's sort of allowed)
Stats uses $context
Who uses $context and $scripturl

**Comment by Oldiesmann** [comment]
I'll look into this when I get a chance. Most of those should be easy to fix, though I'm not entirely sure that the use of $context is an issue.

**Comment by Arantor** [comment]
There is no reason to have any variables in the language files.

**Comment by MissAllSunday** [comment]
So... do we want to use str_replace where those txt vars are used? for example {scripturl} => $scripturl ? There are weird cases like $txt['attach_dir_files_missing'] that are attached to sprintf functions too. Not difficult to add yet another param but it is pretty difficult to track em down.

**Comment by Arantor** [comment]
I do like the idea of having a more descriptive style than sprintf, but I'd then have to suggest going through making everything that style, since having multiple styles going on at once is pretty icky :cry: 

sprintf would be my preference in that respect simply because there's already a bunch of sprintf in place and so it's probably best just to go that way.

**Comment by MissAllSunday** [comment]
Some ideas on this: http://www.simplemachines.org/community/index.php?topic=175031.0

The problem with printf is that it is used for a very large variety of ways all over SMF, normalize all of them its going to be a true nightmare.

**Comment by wintstar** [comment]
How about replace the existing language system with gettext => http://www.gnu.org/software/gettext/gettext.html. It would have some advantages. I am thinking of the plural system and simple translation for the translator. And there might be to edit a module in the admin area as until now.

**Comment by Arantor** [comment]
The plural system issue was documented with a workable solution 18 months ago. The admin area also has an editing area, plus translators already have a versioned system available to them for their language on sm.org meaning that there really is no benefit to going gettext from that perspective.

Add to that, the main problem with gettext is that a lot of hosts still don't have it, meaning you have to deploy it manually which is much slower than what is currently available. Additionally it would mean modders would have to learn how to create .mo and .po files in order to translate their files.

In short: a massive amount of work for perceived-but-never-actualised benefits.

**Comment by frandominguezl** [comment]
What's the final decission on this? @Oldiesmann

**Comment by Arantor** [comment]
Closed as apparently not important enough to fix.
